src/main/scala/com/zeotap/data/io/common/types/SupportedFeatures.scala

```diff
@@ -54,6 +54,13 @@ object SupportedFeatures {
 
   final case class LatestPaths[A](pathTemplate: String, parameters: Map[String, String], relativeToCurrentDate: Boolean) extends SupportedFeatures[A]
 
+  /*
+  This feature tries to handle the cases when there is single partitioned data is given as input,
+  and the computation occurs on a single node only. Objective is before loading such kind of data for computation
+  firstly we would partition the data and write it onto some path, then load this partitioned data and proceed with the computation.
+   */
+  final case class DistributedLoad[A](numberOfPartitions:Option[Int],intermediatePath:String,prioritiseIntermediatePath:Option[Boolean]) extends SupportedFeatures[A]
```

Comment:
Problem : 
Custom handling for Single File Issue - a case where only one large data file is present (say 10gb) and consequently would only load the data on one executor (apache spark) and try all computations on that node essentially losing all advantages of a distributed framework. Feature not yet tested for Beam (but we are pretty sure would occur for any such system)

Feature : 
1. For Apache Spark, resolution is before loading such kind of data for computation,
we first partition the data and write it onto some intermediate path, then load this partitioned data and proceed with the computation.
2. We silently always load the data from this intermediate path if it is present (We determine presence using _SUCCESS file)
3. If not present, we always partition the data into this intermediate path and consequently load from there
*/